crates/driver/src/domain/competition/solution/mod.rs

```diff
@@ -65,7 +65,66 @@ impl Solution {
         weth: eth::WethAddress,
         gas: Option<eth::Gas>,
         fee_handler: FeeHandler,
+        surplus_capturing_jit_order_owners: &HashSet<eth::Address>,
     ) -> Result<Self, error::Solution> {
+        // Surplus capturing JIT orders behave like Fulfillment orders. They capture
+        // surplus, pay network fees and contribute to score of a solution.
+        // To make sure that all the same logic and checks get applied we convert them
+        // right away.
+        for trade in &mut trades {
```

Comment:
I tried doing it there but this could introduce multiple possible issues that would result in a bigger change:
1. the order conversion is a fallible operation which will result in a bunch of error handling code which is more annoying there than in the extra loop.
2. we need to ensure that there is a ucp for all cow amm orders. This becomes way more verbose if we do it in the existing loop but becomes free when it's done in the extra loop.
3. if we so it in that loop we'll not convert the order for solvers that handle fees themselves.

Since this is a hack anyway and should be refactored ASAP I would prefer to merge the code that is the least invasive change that introduces the fewest unforeseen edge cases.